TwitchLib.Client/TwitchClient.cs

```diff
@@ -976,7 +976,7 @@ private Task _client_OnMessage(object sender, OnMessageEventArgs e)
                 _logger?.LogReceived(line);
 
                 _ = OnSendReceiveData?.TryInvoke(this, new() { Direction = SendReceiveDirection.Received, Data = line });
-                return HandleIrcMessageAsync(IrcParser.ParseMessage(line));
+                _ = HandleIrcMessageAsync(IrcParser.ParseMessage(line));
```

Comment:
Since it's in a for-each, don't you want to properly `await` it? Why just "fire & forget"?

Reply:
`await`ing would cause us to handle events sequentially rather than concurrently, and if the "prologue" part of any method down the callstack throws, we will still get the exception here. Do you have any concerns regarding ordering?

Reply:
@Bukk94 Actually, I thought about transforming lines into a select consumed by Task.WhenAll but that would be, relatively speaking, really expensive compared to now very streamlined even handling pipeline.

The difference is right now we only catch exception before the task code yields back to `_client_OnMessage` method while with `await` we would catch exceptions after the yield (e.g. some network call in user handler code) too. This might be a bit subtle yet complex detail, and I'm not sure whether the users would want the exception to propagate all the way out of their handler, back into the twitch client's handler registered with`IClient`, and so on and so forth.

Reply:
Well I think that in current state it might cause some issues. 
Both `OnSendReceiveData?.TryInvoke` (btw there shouldn't be null check) and `HandleIrcMessageAsync` are `Task`s. What I would expect is that first, I'll get OnSendRecieveData event and then I'll get appropriate event from parser/handler (e.g. privmsg event).
With parallel processing like this we have no control over the sequence. If you start processing multiple lines at once, it's not guaranteed that it will be exactly as it was read by the socket. I think this is highly unwanted behavior and we should rather await it and force synchronicity. It will be slower, yes, but at least it will ensure correct lines order. WDYT?

Reply:
`OnSendReceiveData` was never awaited. I'm fine with enforcing line order, just not sure how relevant this is though, the issue is it will (asynchonously) block the handler execution and whatever implementation awaits `IClient.OnMessage` won't be able to advance until user handler code runs (which is how it was before rewrite to async, but still).

Reply:
It was never awaited because previously both methods were `void`. It changed with our recent changes. We changed `OnSendReceiveData` from `EventHandler` to `AsyncEventHandler` and (probably I) changed `HandleIrcMessageAsync` to be Task (previously also `void`).
But you have a point about blocking that event handler. Do you think it was bad decision to implement `AsyncEventHandler` and we should go back to standard event handlers?

Reply:
Yes, I know, but it was also not awaited after event handlers were made async (most likely accidentally). The better way to handle this while ensuring line ordering is to have bounded `Channel` with some reasonable limit, which will allow the WebSocket client implementation to advance, pushing the unhandled messages to channel, while the client event handling code catches up, and if it doesn't, only then block the client too. I'll look into what we can do here.